packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieve.ts

```diff
@@ -56,12 +57,69 @@ export class ManifestOrSourcePathGatherer
       // corresponding to the explorer path when we begin supporting multiple workspace folders
       const workspaceRootPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
       const manifestPath = path.join(workspaceRootPath, 'manifest');
-      const isManifestFile = this.explorerPath.includes(manifestPath);
-      const type = isManifestFile ? FileType.Manifest : FileType.Source;
-      return { type: 'CONTINUE', data: { filePath: this.explorerPath, type } };
+      const isManifest = this.explorerPath.includes(manifestPath);
+      const type = isManifest ? FileType.Manifest : FileType.Source;
+
+      // If the type is manifest check whether or not the user selected an actual
```

Comment:
No need for this comment, code is clear enough to follow.

Reply:
Sure, why not. The next person to touch this code can just reason out what's going on rather than reading the intent.

Reply:
If you think the code itself doesn't communicate intent then change it so it does and we don't have to add this type of comments.